### PR TITLE
feat: add user update and delete route

### PR DESCRIPTION
**Pull request by abhi-yo**

### CODE REVIEW

File: server/bun.lockb
Comment (sm-sami) [inline server/bun.lockb:1]: why changes to this file with no changes to package.json 🤔

File: server/src/routes/user.ts [hunk 1]
```diff
@@ -125,12 +114,63 @@ user.get("/me", authenticateUser, async (c) => {
...
     .where(eq(users.email, email));
-
   if (!user) {
     return c.json({ success: false, error: "User not found" }, 404);
   }
-
   return c.json({ success: true, data: { user } });
 });
 
+user.put("/update", authenticateUser, zValidator("json", UpdateUserSchema), async (c) => {
```
Comment (sm-sami) [inline server/src/routes/user.ts]: should be a patch route also it can just be `/`

File: server/src/routes/user.ts [hunk 2]
```diff
@@ -125,12 +114,63 @@ user.get("/me", authenticateUser, async (c) => {
...
+  const { email: currentEmail } = c.get("jwtPayload");
+  const updateData = c.req.valid("json");
+
+  try {
+    const [updatedUser] = await db
+      .update(users)
+      .set({
+        ...updateData,
+        updatedAt: new Date(),
```
Comment (sm-sami) [inline server/src/routes/user.ts:132]: `updateAt` will be changed automatically, it is declared in the schema, can you check and confirm

File: server/src/routes/user.ts [hunk 3]
```diff
@@ -125,12 +114,63 @@ user.get("/me", authenticateUser, async (c) => {
...
+      message: "User updated successfully",
+      data: { user: updatedUser },
+    });
+  } catch (error) {
+    console.error("Update error:", error);
+    return c.json({ success: false, error: "Failed to update user" }, 500);
+  }
+});
+
+user.delete("/delete", authenticateUser, async (c) => {
```
Comment (sm-sami) [inline server/src/routes/user.ts]: the route can just be `/`, route method gives enough information already

Comment (sm-sami) [inline server/src/routes/user.ts]: the input is not validated

File: server/src/schema/users.ts [hunk 4]
```diff
@@ -42,11 +44,40 @@ export const SignInUserSchema = UserSchema.pick({
...
+    .length(15, { message: "Registration number must be 15 characters" })
+    .optional(),
+  password: z
+    .string()
+    .min(8, {
+      message: "Password must be at least 8 characters",
+    })
+    .optional(),
+}).refine((data) => Object.keys(data).length > 0, {
+  message: "At least one field must be provided for update",
```
Comment (sm-sami) [inline server/src/schema/users.ts]: why this change? the user must not be able to change anything except name and password

File: server/src/schema/users.ts [hunk 5]
```diff
@@ -42,11 +44,40 @@ export const SignInUserSchema = UserSchema.pick({
...
 });
 
-export const DeleteUserSchema = UserSchema.pick({
-  email: true,
+export const DeleteUserSchema = z.object({
+  password: z.string({ required_error: "Password is required for account deletion" }),
```
Comment (sm-sami) [inline server/src/schema/users.ts:63]: why?

File: server/src/schema/users.ts [hunk 6]
```diff
@@ -42,11 +44,40 @@ export const SignInUserSchema = UserSchema.pick({
...
 });
+
+export type User = z.infer<typeof UserSchema>;
+export type SignInUser = z.infer<typeof SignInUserSchema>;
+export type UpdateUser = z.infer<typeof UpdateUserSchema>;
+export type DeleteUser = z.infer<typeof DeleteUserSchema>;
```
Comment (sm-sami) [inline server/src/schema/users.ts]: are these being used?

**Review by sm-sami — CHANGES_REQUESTED** [review]

File: server/src/schema/users.ts (hunk 4 above)
Comment (abhi-yo) [reply server/src/schema/users.ts]: image

File: server/bun.lockb
Comment (abhi-yo) [reply server/bun.lockb:1]: I am not aware of that
